### PR TITLE
Cleaning up old code to prevent warnings

### PR DESCRIPTION
**Pull request by renatoh**
Reading through the code in org.apache.solr.cli, I realized there are quite some warnings, which can be easily cleaned up

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cli/SimplePostTool.java [hunk 1]
```diff
@@ -544,14 +537,20 @@ public int postFiles(File[] files, int startIndexInArgs, OutputStream out, Strin
...
     for (File srcFile : files) {
-      boolean isValidPath = checkIsValidPath(srcFile);
-      if (isValidPath && srcFile.isDirectory() && srcFile.canRead()) {
-        filesPosted += postDirectory(srcFile, out, type);
-      } else if (isValidPath && srcFile.isFile() && srcFile.canRead()) {
-        filesPosted += postFiles(new File[] {srcFile}, out, type);
-      } else {
-        filesPosted += handleGlob(srcFile, out, type);
-      }
+      filesPosted = getFilesPosted(out, type, filesPosted, srcFile);
```
[the remark this answers is not in the document]
Comment (renatoh) [reply solr/core/src/java/org/apache/solr/cli/SimplePostTool.java:540]: true, 'out' passed in to getFilesPosted is always 0 hence it does not need be posted in. will fix it. thanks

File: solr/core/src/java/org/apache/solr/cli/AssertTool.java [hunk 2]
```diff
@@ -30,6 +30,7 @@
 import org.apache.solr.client.solrj.SolrClient;
 import org.apache.solr.client.solrj.SolrServerException;
 import org.apache.solr.client.solrj.request.HealthCheckRequest;
+import org.apache.solr.client.solrj.response.CollectionAdminResponse;
```
[the remark this answers is not in the document]
Comment (renatoh) [reply solr/core/src/java/org/apache/solr/cli/AssertTool.java:33]: nope, not needed anymore, will remove the unused imports.

**Comment by epugh** [comment]
@renatoh can you check the precommit failure?   It's a weird one where we can't do waht the warning tells us to do....   Please ping me if I don't get this merged after you get precommit fixed.

**Comment by renatoh** [comment]
I am a bit confused, so why would we need an assert at a place where it is impossible for the Socket to be null?

**Comment by epugh** [comment]
> I am a bit confused, so why would we need an assert at a place where it is impossible for the Socket to be null?

Because otherwise precommit fails ;-).   We really should figure out why Precommit fails, and get the underlying problem fixed, maybe for another PR?.  I'm going to go ahead and get this merged as tests passed locally for me.

**Comment by epugh** [comment]
Thanks for these updates!  For staight up code refactorings we don't normally do a CHANGES.txt entry, but I very much appreciate the contribution.

**Comment by renatoh** [comment]
your are welcome. looking at the source code I am seeing a lot of potential for small refactoring.
most of them just to clean up the code and reduce the number of warnings, but some of them could even give a small performance benefit.
if this kind of refactoring is appreciated I am happy to do some more:)

**Comment by epugh** [comment]
Please do!  The improvement in the 9x line (as well as main) to the code quality has been huge, but there is so much more to do.

**Pull request merged**